services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceParentHelixAdmin.java

```diff
@@ -1846,6 +1846,50 @@ public void testUpdateStoreNativeReplicationSourceFabric() {
         "Native replication source fabric does not match after updating the store!");
   }
 
+  @Test
+  public void testDisableHybridConfigWhenAAandIncPushConfigIsEnabled() {
+    String storeName = Utils.getUniqueString("testUpdateStore");
+    Store store = TestUtils.createTestStore(storeName, "test", System.currentTimeMillis());
+
+    store.setHybridStoreConfig(
+        new HybridStoreConfigImpl(
+            1000,
+            100,
+            -1,
+            DataReplicationPolicy.NON_AGGREGATE,
+            BufferReplayPolicy.REWIND_FROM_EOP));
+    store.setActiveActiveReplicationEnabled(true);
+    store.setIncrementalPushEnabled(true);
+    store.setChunkingEnabled(true);
+    doReturn(store).when(internalAdmin).getStore(clusterName, storeName);
+
+    doReturn(CompletableFuture.completedFuture(new SimplePubSubProduceResultImpl(topicName, partitionId, 1, -1)))
+        .when(veniceWriter)
+        .put(any(), any(), anyInt());
+
+    when(zkClient.readData(zkMetadataNodePath, null)).thenReturn(null)
+        .thenReturn(AdminTopicMetadataAccessor.generateMetadataMap(1, -1, 1));
+
+    parentAdmin.initStorageCluster(clusterName);
+    parentAdmin.updateStore(
```

Comment:
Thanks a lot! Can you also add a few lines to confirm exceptions would be thrown as expected?

Reply:
I want to make sure it would fail loudly and would not be removed by someone accidentally

Reply:
What do you mean by this? before the logic change, the Assertion will be thrown at the end because it won't disable hybrid config as expected. With updated logic, new store is no longer hybrid at the end of the test.

Reply:
Sure. I was asking if you can add a few more tests for unhappy path where the exceptions would be thrown and we verify that.

Reply:
Do we have tests to exercise line 2319 2325?

Reply:
Oh I see what you mean. This part I can add it

Reply:
Updated, could you please check again?